### PR TITLE
Realsense nodelet crash - nodelet::Loader::Impl

### PR DESCRIPTION
**Pull request by guillaumeautran**
The `nodelet::detail::CallbackQueueManager::removeQueue` removes an entry from the list of queues without checking first that the entry exists in the queue. When that happens, the underlying C++ library throws an object not found exception aborting the process.
This change only adds a test to make we only remove a queue that exists.

### CODE REVIEW

File: nodelet/src/callback_queue_manager.cpp
Comment (efernandez) [inline nodelet/src/callback_queue_manager.cpp:124]: This `ROS_ASSERT` has the same condition you're testing in the new `if` below. Not a big deal, but I'd suggest removing the `ROS_ASSERT` in the same commit that adds the `if`:

```suggestion
```

Strictly speaking, `removeQueue` should never get called with a `queue` that isn't in the `queues_`. Do you know why that could have happened?

**Review by efernandez — CHANGES_REQUESTED** [review]
LGTM, but I wonder if we could avoid calling `removeQueue` when the `queue` isn't in the `queues_` instead. I'm just trying to have a better understanding of what could have happened.

File: nodelet/src/callback_queue_manager.cpp
Comment (guillaumeautran) [reply nodelet/src/callback_queue_manager.cpp:124]: No, I have not been able to figure out why it is being called with a queue not in the queues_ list. It does not make sense because the argument passed in should never have been removed. https://github.com/clearpathrobotics/nodelet_core/blob/indigo-devel/nodelet/src/loader.cpp#L197

Comment (efernandez) [reply nodelet/src/callback_queue_manager.cpp:124]: Do you have a coredump showing the addresses of `st_queue` and `mt_queue`?

https://github.com/clearpathrobotics/nodelet_core/blob/a7da9266c766f01cfe667363f00104d7cfcb5f68/nodelet/src/loader.cpp#L184-L185

I wonder if both are set, and different. The only case I can think of that could lead to `queues_` not having one of them is if both `st_queue` and `mt_queue` are `NULL`. In that case, `addQueue` would add both queues to the `queues_` entry `0` (`NULL`), so `removeQueue` would fail in the second call: https://github.com/clearpathrobotics/nodelet_core/blob/a7da9266c766f01cfe667363f00104d7cfcb5f68/nodelet/src/loader.cpp#L198

Comment (efernandez) [reply nodelet/src/callback_queue_manager.cpp:124]: I looked at a core dump related to this and `st_queue != mt_queue != NULL`. The crash also happens in the first call to `removeQueue`, and it comes from the loader running in https://github.com/clearpathrobotics/nodelet_core/blob/indigo-devel/nodelet/src/loader.cpp#L294, which suggests the pluginlib subsystem was having trouble to find the nodelet plugin name. Maybe there's a corner case when that code runs, which leads to this.

Either way, I'm approving this for now as is, since I don't have a better solution atm.

Let me know if you don't have merge permissions, and I'd merge this.

**Review by efernandez — APPROVED** [review]

**Review by servos — APPROVED** [review]

**Pull request merged**
